lib/engine/graph.rb

```diff
@@ -89,6 +89,69 @@ def reachable_hexes(corporation)
       @reachable_hexes[corporation]
     end
 
+    # How far is the start_hex from one of the corporation's stations?
+    # This is by track, not as the crow flies.
+    # start_cy is either a City or nil (plain track or town)
+    #
+    # This implementation assumes:
+    #  - Lawson Track
+    #  - No double dits
+    #  - OO tiles should work? NY tile works
+    def distance_to_station(corporation, start)
```

Comment:
can you just use node.walk or path.walk to generate all the possible paths

these assumptions you made shouldn't be necessary and it's weird that you implemented your own bfs

Reply:
I couldn't figure out how `node.walk` or `path.walk` work or what the inputs do, but since they looked recursive I figured it used a DFS. 

Since I needed a BFS to find the shortest path I tried make a simple one for 18FL, which leads to the assumptions which are cases that aren't handled in the implementation

Reply:
i'm not satisfied with this implementation, i think we need to put some more thought into this to do in a more robust manner.

Reply:
let's chat on slack to figure out the best algorithm / approach to make this generically useful